packages/valory/skills/twitter_scoring_abci/behaviours.py

```diff
@@ -123,38 +124,96 @@ class TwitterCollectionBehaviour(TwitterScoringBaseBehaviour):
 
-            # Get mentions from Twitter
-            tweets, latest_mention_tweet_id = yield from self._get_twitter_mentions()
+        # Window has expired
+        if current_time >= last_tweet_pull_window_reset + ONE_DAY:
+            return False, 0, current_time
 
-            # Get hashtags from Twitter
-            if tweets != TwitterCollectionRound.ERROR_PAYLOAD:
-                (
-                    tweets,
-                    latest_hashtag_tweet_id,
-                ) = yield from self._get_twitter_hashtag_search(tweets)
+        # Reached max number of tweets
+        if number_of_tweets_pulled_today >= self.params.max_tweet_pulls_allowed:
+            return True, number_of_tweets_pulled_today, last_tweet_pull_window_reset
 
-                # Keep the max latest_tweet_id
-                valid_latest_ids = [
-                    i for i in (latest_mention_tweet_id, latest_hashtag_tweet_id) if i
-                ]
-                if valid_latest_ids:
-                    latest_mention_tweet_id = max(valid_latest_ids)
+        # Window has not expired and we have not reached the max number of tweets
+        return False, number_of_tweets_pulled_today, last_tweet_pull_window_reset
+
+    def async_act(self) -> Generator:
+        """Do the act, supporting asynchronous execution."""
+
+        with self.context.benchmark_tool.measure(self.behaviour_id).local():
 
-            if tweets == TwitterCollectionRound.ERROR_PAYLOAD:
-                payload_data = TwitterCollectionRound.ERROR_PAYLOAD
+            (
+                has_limit_reached,
+                number_of_tweets_pulled_today,
+                last_tweet_pull_window_reset,
+            ) = self._check_daily_limit()
+            if has_limit_reached:
+                payload = TwitterCollectionRound.ERROR_PAYLOAD
             else:
-                self.context.logger.info(
-                    f"Retrieved new tweets [until_id={latest_mention_tweet_id}]: {list(tweets.keys())}"
+                # Get mentions from Twitter
+                (
+                    tweets_0,
+                    latest_mention_tweet_id,
+                    number_of_tweets_pulled_today,
+                ) = yield from self._get_twitter_mentions(
+                    number_of_tweets_pulled_today=number_of_tweets_pulled_today
                 )
-                payload_data = {
-                    "tweets": tweets,
-                    "latest_mention_tweet_id": latest_mention_tweet_id,
-                }
+
+                # Get hashtags from Twitter
+                if tweets_0 != TwitterCollectionRound.ERROR_PAYLOAD:
+                    (
+                        tweets_1,
+                        latest_hashtag_tweet_id,
+                        number_of_tweets_pulled_today,
+                    ) = yield from self._get_twitter_hashtag_search(
+                        tweets_0,
+                        number_of_tweets_pulled_today=number_of_tweets_pulled_today,
```

Comment:
Be aware that here we pass the tweet dict that we got from the mention call to the method that makes the recent search call, and this second function adds the new tweets to the existing ones. So in this case I think tweets_1 also contains tweets_0.